be/CMakeLists.txt

```diff
@@ -989,8 +989,18 @@ set (TEST_LINK_LIBS ${STARROCKS_LINK_LIBS}
 # Only build static libs
 set(BUILD_SHARED_LIBS OFF)
 
-if (NOT $ENV{STARROCKS_LINKER} STREQUAL "")
-    add_link_options("-fuse-ld=$ENV{STARROCKS_LINKER}")
+set (STARROCKS_LINKER "$ENV{STARROCKS_LINKER}")
+include(cmake_modules/FindGlibcVersion.cmake)
+
+if (GLIBC_VERSION VERSION_LESS "2.29" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 14.0.0)
+    if (NOT DEFINED ${STARROCKS_LINKER} OR ${STARROCKS_LINKER} STREQUAL "")
+        set(STARROCKS_LINKER "gold")
```

Comment:
is it `gold` or `ld.gold`? didn't find `gold` in our dev-env-centos7 docker image.

better add message to say that the linker is forcibly overwritten to `ld.gold`.

Reply:
gold

Reply:
```
-fuse-ld=gold
```
or
```
-B/usr/local/bin/ld.gold
```